src/main/java/com/patchworkmc/PatchworkUI.java

```diff
@@ -558,37 +557,21 @@ private static void startPatching() throws IOException {
 			}
 		}
 
-		NaiveRemapper naiveRemapper = new NaiveRemapper(bridged);
 		LOGGER.info("Preparation Complete!\n");
 
-		File inputFolder = new File(modsFolder.getText());
+		Path inputFolder = new File(modsFolder.getText()).toPath();
 		Path outputFolder = new File(PatchworkUI.outputFolder.getText()).toPath();
-		int[] patched = {0};
+		Path tempFolder = Files.createTempDirectory(rootPath, "temp");
+		Patchwork patchwork;
 
-		Files.walk(inputFolder.toPath()).forEach(path -> {
-			if (!path.toString().endsWith(".jar")) {
-				return;
-			}
-
-			String modName = path.getFileName().toString().replaceAll(".jar", "");
-			LOGGER.info("=== Patching " + path.toString() + " ===");
-
-			try {
-				Patchwork.transformMod(rootPath, path, outputFolder, modName, bridged, naiveRemapper);
-
-				if (yarnBuild != null) {
-					LOGGER.info("Remapping " + modName + " (intermediary -> yarn)");
-					Patchwork.remap(yarnMappings[0], outputFolder.resolve(modName + ".jar"), outputFolder.resolve(modName + "-dev.jar"), rootPath.resolve("data/" + version + "-client+intermediary.jar"));
-				}
-
-				patched[0]++;
-			} catch (Throwable t) {
-				LOGGER.error("Transformation failed, skipping current mod!");
+		if (generateDevJar.isSelected()) {
+			patchwork = new Patchwork(inputFolder, outputFolder, rootPath.resolve("data/"), tempFolder, bridged, yarnMappings[0]);
```

Comment:
If we did the change I suggested above, this could be deduplicated, perhaps using Collections.emptyList and Collections.singletonList... maybe use an Optional instead?

Reply:
I can't find a way that's cleaner than just an if/else statement.

Reply:
Wait, I didn't read the List suggestion above. I'll take another look.